client/my-sites/theme/main.jsx

```diff
@@ -436,10 +438,14 @@ const ThemeSheet = React.createClass( {
 	},
 
 	getDefaultOptionLabel() {
-		const { defaultOption, isActive, isLoggedIn, isPremium, isPurchased } = this.props;
+		const { defaultOption, isActive, isLoggedIn, isPremium, isPurchased, isJetpack } = this.props;
 		if ( isLoggedIn && ! isActive ) {
 			if ( isPremium && ! isPurchased ) { // purchase
-				return i18n.translate( 'Pick this design' );
+				return config.isEnabled( 'jetpack/pijp' ) && isJetpack
+					? i18n.translate( 'Upgrade to activate', {
+						comment: 'label prompting user to upgrade the Jetpack plan to activate a certain theme'
+					} )
+					: i18n.translate( 'Pick this design' );
```

Comment:
I think you can drop this logic if you update the `defaultOption` picking logic instead to use `upgradePlan` when applicable.